scripts/localdev.py

```diff
@@ -970,21 +970,81 @@ def build_scilla(config, driver, tag):
                capture_output = False)
     return image_name
 
+def build_native_scilla(config):
+    build_env = os.environ.copy()
+    build_env.update(config.default_env)
+
+    if not build_env.get("VCPKG_ROOT"):
+        raise GiveUp("Environment variable VCPKG_ROOT must be defined to point to vcpkg")
+
+    build_env["SCILLA_REPO_ROOT"] = SCILLA_DIR
+
+    # Cleanup
+    shutil.rmtree(os.path.join(SCILLA_DIR, "vcpkg_installed"), ignore_errors = True)
+    shutil.rmtree(os.path.join(SCILLA_DIR, "_build"), ignore_errors = True)
+
+    output = run_or_die(config, ["grep", "OCAML_VERSION_RECOMMENDED=", "Makefile"], in_dir = SCILLA_DIR, env = build_env,
+               capture_output = True).decode('utf-8')
+    result = re.match(r'^OCAML_VERSION_RECOMMENDED=([0-9\.]+)$', output)
+    if len(result.groups()) != 1:
+        raise GiveUp("Couldn't infer OCaml recommended version for Scilla")
+
+    # Build OCaml
+    run_or_die(config, ["opam", "init", f"--compiler=ocaml-base-compiler.{result.group(1)}", "--yes"], in_dir = SCILLA_DIR, env = build_env,
+               capture_output = False)
+    run_or_die(config, ["./scripts/build_deps.sh"], in_dir = SCILLA_DIR, env = build_env,
+               capture_output = False)
+
+    for p in map(
+        # Skip the 'export ' and split at '=' into a tuple
+        lambda x: x.split('='),
+        filter(
+            lambda x: x.find('export') == -1,
+            re.split(";\n?",
+            run_or_die(config, ["opam", "env"], capture_output=True).decode('utf-8')))):
+        if len(p) > 1:
+            build_env[p[0]] = p[1][1:-1]
+
+    # Set PWD explicitly to get around usage in Scilla's Makefile; in sub-shells
+    # this will resolve to the wrong directly which could mess up rpath.
+    build_env["PWD"] = SCILLA_DIR
+    run_or_die(config, ["make", "opamdep"], in_dir = SCILLA_DIR, env = build_env,
+               capture_output = False)
+    run_or_die(config, ["make"], in_dir = SCILLA_DIR, env = build_env,
+               capture_output = False)
```

Comment:
OK, but why is this here and not in the makefile? I think I'm missing something .. ?

Reply:
These are in the Makefile but they're separate target and apart from the Makefile being fairly old (and at least partially out-of-date?), I find that something there & in the instructions is very sensitive to the order in which this is done.
So I was hoping to do this independently for now without having this depend on other PRs in Scilla. Once stable, we will fix Scilla & the instructions.